### ISSUE TITLE
NSEC records with parameter Types 

### ISSUE THREAD

**Issue by victorkani** [richardschneider/net-mdns#27]
How to send answer of Record type NSEC  with types TXT and SRV.

Something like this , I found this in ARSoft.Tools.Net
public NSecRecord(string name, RecordClass recordClass, int timeToLive, string nextDomainName, List<RecordType> types): base(name, RecordType.NSec, recordClass, timeToLive)
{
 }

**Comment by richardschneider** [comment]
You can add any  [DNS Resource Record](https://richardschneider.github.io/net-dns/articles/rr.htmlhttps://richardschneider.github.io/net-dns/articles/rr.html) to  `ServiceProfile.Resources`.  Then, when the Resource Record is asked for, it will be sent.

```csharp
var service = new ServiceProfile("x1", "_xservice._tcp", 5011);
service.Resources.Add(new NSecRecord { ... } );
var sd = new ServiceDiscovery();
sd.Advertise(servuce));

```

**Comment by richardschneider** [comment]
The [NSecRecord](https://richardschneider.github.io/net-dns/api/Makaretu.Dns.NSECRecord.html) can be created as

```csharp
new NSecRecord 
{
   NextOwnerName = "...",
   TypeBitmaps = ,..
}
```

**Comment by richardschneider** [comment]
@victorkani Does the above answers your question?

**Comment by victorkani** [comment]
How to return the following records to google discovery service ?

 The following records should be returned by the server in an authoritative response:
Two NSEC records, the first for the name {friendlyName}._googlecast._tcp.local with types TXT and SRV, the second for name {friendlyName}.local with type A. Where {friendlyName}is the display name of the device

                            res.AdditionalRecords.Add(new NSECRecord()
                            {
                                Name = "victor." + service,
                                Type = DnsType.TXT,
                                NextOwnerName = "1.victor.local",

                            });
The above is the syntax for sending NSEC record information, but the Type parameter is not of array type to set both TXT and SRV.

**Comment by richardschneider** [comment]
I'm just making a change to Makaretu.Dns to make  `NSECRecord` friendlier, see https://github.com/richardschneider/net-dns/issues/17.  In the current design you need to set `TypesBitmap` which is byte array and way unfriendly.

In the new design
```csharp
new NSECRecord()
{
      Name = "victor." + service,
      Type = { DnsType.TXT, DnsType.SRV },
      NextOwnerName = "1.victor.local"
}
```

I'll have a release in a few hours and will ping you.

**Comment by richardschneider** [comment]
The new NSECRecord is available in release [v0.12.0](https://www.nuget.org/packages/Makaretu.Dns.Multicast/0.12.0)

Hope it helps!

**Comment by richardschneider** [comment]
@victorkani Are you ready to close this issue?
